akka-projection-cassandra/src/test/scala/akka/projection/cassandra/CassandraProjectionSpec.scala

```diff
@@ -519,19 +517,16 @@ class CassandraProjectionSpec
     "verify offsets before processing an envelope" in {
       val entityId = UUID.randomUUID().toString
       val projectionId = genRandomProjectionId()
-      val (verifiedQueue, verifiedProbe) = Source
-        .queue[Long](1, OverflowStrategy.backpressure)
-        .toMat(TestSink.probe(system.classicSystem))(Keep.both)
-        .run()
+      val verified = ListBuffer[Long]()
```

Comment:
I don't think ListBuffer is thread safe. Might be ok in the way we have currently implemented things but making to much assumptions of implementation details that the test should not know about. For example, if we would have an async boundary between the verification and the handler. Might be better to add the verified offsets to a thread safe Set and check that it contains the offset from the handler.

Reply:
Ok :+1:

Reply:
I'm using a test probe there like you originally suggested.